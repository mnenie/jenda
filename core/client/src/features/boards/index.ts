export * from "./add-board/ui";
export * from "./filter/ui";
