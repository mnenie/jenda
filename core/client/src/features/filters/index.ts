export * from './boards-filters/ui'
