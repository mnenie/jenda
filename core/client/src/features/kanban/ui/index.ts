import { AddColumn } from "./add-column";
import ChangeName from "./ChangeName.vue";
import ChooseStatus from "./ChooseStatus.vue";
import DragCards from "./DragCards.vue";
import DragColumns from "./DragColumns.vue";
import FilterTasks from "./FilterTasks.vue";
import RemoveBoard from "./RemoveBoard.vue";
import UpdateBoard from "./UpdateBoard.vue";

export {
  AddColumn,
  ChangeName,
  ChooseStatus,
  DragCards,
  DragColumns,
  FilterTasks,
  RemoveBoard,
  UpdateBoard,
};
