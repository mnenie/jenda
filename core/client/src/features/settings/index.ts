export * from "./lang-switcher";
export * from "./theme-switcher";
