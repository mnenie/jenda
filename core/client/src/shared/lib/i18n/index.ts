import { useCookies } from '@vueuse/integrations/useCookies'
import { computed } from 'vue'
import { createI18n } from 'vue-i18n'
import enLocale from './locales/en-US'
import ruLocale from './locales/ru-RU'
import zhLocale from './locales/zh-CN'

type MessageSchema = typeof enLocale

const messages = {
  'en-US': {
    ...enLocale,
  },
  'ru-RU': {
    ...ruLocale,
  },
  'zh-CN': {
    ...zhLocale,
  },
}
const cookies = useCookies()

const getCurrentLocale = computed(() => {
  const cookieLanguage = cookies.get('i18n')
  if (cookieLanguage) {
    return cookieLanguage
  }
  if (navigator.language.split('-')[0] === 'ru') {
    return 'ru-RU'
  }
  return 'en-US'
})

const i18n = createI18n<[MessageSchema], 'en-US' | 'ru-RU' | 'zh-CN'>({
  legacy: false,
  locale: getCurrentLocale.value,
  globalInjection: true,
  messages,
})

export default i18n
