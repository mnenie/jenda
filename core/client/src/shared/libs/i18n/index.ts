import { computed } from 'vue'
import { createI18n } from 'vue-i18n'

import en from '../vee-validate/rules/en'
import ru from '../vee-validate/rules/ru'
import zh from '../vee-validate/rules/zh'
import enLocale from './locales/en-US'
import ruLocale from './locales/ru-RU'
import zhLocale from './locales/zh-CN'

export type MessageSchema = typeof enLocale

const messages = {
  'en-US': {
    ...enLocale,
    errors: {
      ...en,
    },
  },
  'ru-RU': {
    ...ruLocale,
    errors: {
      ...ru,
    },
  },
  'zh-CN': {
    ...zhLocale,
    errors: {
      ...zh,
    },
  },
}

const getCurrentLocale = computed(() => {
  const storageLanguage = localStorage.getItem('i18n')
  if (storageLanguage) {
    return storageLanguage
  }
  if (navigator.language.split('-')[0] === 'ru') {
    return 'ru-RU'
  }
  return 'en-US'
})

const i18n = createI18n<[MessageSchema], 'en-US' | 'ru-RU' | 'zh-CN'>({
  legacy: false,
  locale: getCurrentLocale.value,
  globalInjection: true,
  messages,
})

export default i18n
