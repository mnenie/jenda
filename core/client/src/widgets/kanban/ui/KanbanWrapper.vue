<script setup lang="ts">
import type { Column } from '@/entities/board'
import { AddColumn, DragCards, DragColumns } from '@/features/kanban'

defineProps<{
  columns?: Column[]
}>()
</script>

<template>
  <div :class="$style.kanban_wrapper">
    <div v-if="columns">
      <DragColumns :columns="columns">
        <template #default="{ cards }">
          <DragCards :cards="cards" />
        </template>
      </DragColumns>
    </div>
    <AddColumn />
  </div>
</template>

<style module lang="scss">
.kanban_wrapper {
  padding: 15px;
  bottom: 50px;
  left: 0;
  overflow-x: auto;
  overflow-y: hidden;
  position: absolute;
  right: 0;
  scrollbar-color: rgba(var(--zinc-rgb-300), 0.3);
  scrollbar-width: auto;
  top: 0;
  -webkit-user-select: none;
  user-select: none;
  white-space: nowrap;
  display: flex;
  flex-direction: row;
  list-style: none;
  gap: 15px;
}

.flip-list-move {
  transition: transform 0.3s;
}

.no-move {
  transition: transform 0s;
}

.ghost {
  opacity: 0.2;
}
</style>
