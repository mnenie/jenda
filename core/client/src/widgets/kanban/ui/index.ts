import InfoPart from './InfoPart.vue'
import KanbanWrapper from './KanbanWrapper.vue'

export { InfoPart, KanbanWrapper }
