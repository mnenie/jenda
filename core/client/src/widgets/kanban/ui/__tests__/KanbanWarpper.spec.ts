import { AddColumn, DragColumns } from '@/features/kanban'
import i18n from '@/shared/lib/i18n'
import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import KanbanWrapper from '../KanbanWrapper.vue'
import '@/shared/lib/vitest-utils/cookiesI18n-mock'

describe('tests for KanbanWrapper.vue', () => {
  const wrapper = mount(KanbanWrapper, {
    global: { plugins: [i18n] },
    props: {
      columns: [],
    },
  })

  it('should be render correctly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should render subcomponents', () => {
    expect(wrapper.findComponent(DragColumns).exists()).toBe(true)
    expect(wrapper.findComponent(AddColumn).exists()).toBe(true)
  })
})
