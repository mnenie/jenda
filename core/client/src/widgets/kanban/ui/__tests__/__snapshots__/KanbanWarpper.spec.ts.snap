// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`tests for KanbanWrapper.vue > should be render correctly 1`] = `
"<div class="kanban_wrapper">
  <div>
    <div class="columns" tag="ul" name="flip-list" type="transition"></div>
  </div>
  <div class="text-sm block">
    <div i-lucide-plus="" text-base=""></div> Add new column
  </div>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;
